pkg/graveler/ref/db_manager.go

```diff
@@ -502,3 +526,64 @@ func (m *DBManager) addGenerationToNodes(nodes map[graveler.CommitID]*CommitNode
 		nodesCommitIDs = nextIterationNodes
 	}
 }
+
+// rWorker writes a repository, and all repository related entities, to fd
+func rWorker(rChan <-chan *graveler.RepositoryRecord) error {
+	for r := range rChan {
+		pb := graveler.ProtoFromRepo(r)
+		data, err := proto.Marshal(pb)
+		if err != nil {
+			return err
+		}
+		if err = encoder.Encode(kv.Entry{
+			PartitionKey: []byte(graveler.RepoPartition()),
+			Key:          []byte(graveler.RepoPath(r.RepositoryID)),
+			Value:        data,
+		}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func Migrate(ctx context.Context, d *pgxpool.Pool, writer io.Writer) error {
+	encoder = kv.SafeEncoder{
+		Je: json.NewEncoder(writer),
+		Mu: sync.Mutex{},
+	}
+	// Create header
+	if err := encoder.Encode(kv.Header{
+		LakeFSVersion:   version.Version,
+		PackageName:     packageName,
+		DBSchemaVersion: kv.InitialMigrateVersion,
+		CreatedAt:       time.Now().UTC(),
+	}); err != nil {
+		return err
+	}
+
+	rChan := make(chan *graveler.RepositoryRecord, migrateQueueSize)
+	var g multierror.Group
+	for i := 0; i < jobWorkers; i++ {
+		g.Go(func() error {
+			return rWorker(rChan)
+		})
+	}
+
+	rows, err := d.Query(ctx, "SELECT * FROM graveler_repositories")
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+	rowScanner := pgxscan.NewRowScanner(rows)
+	for rows.Next() {
+		r := new(graveler.RepositoryRecord)
+		err = rowScanner.Scan(r)
+		if err != nil {
+			return err
```

Comment:
I think the channel is not closed here, which can result with ungraceful exit of the workers.

Reply:
Done